### ISSUE TITLE
Sorted stream should be collected to a LinkedHashSet instead of a Set

### ISSUE THREAD

**Issue by yeikel** [openrewrite/rewrite#1609]
The following code is incorrect as the returned Set will ignore the sort operation and return the values in any order

```java
               Set<Integer> l = new HashSet<>();
                l.add(5);
                l.add(60);
                l.add(3);

                Set<Integer> sorted = l.stream().sorted().collect(Collectors.toSet());

```

```java


                Set<Integer> l = new HashSet<>();
                l.add(5);
                l.add(60);
                l.add(3);

                Set<Integer> sorted = l.stream().sorted().collect(Collectors.toCollection(LinkedHashSet::new));
```

**Comment by RossZhang-cpu** [comment]
Hi guys, Is this still exist? If yes, may you please assgin to me and I could do some modification.

**Comment by yeikel** [comment]
Yes, as the issue is still open, you can assume that it is up for grabs. I can't assign it to you, but @timtebeek will

You can probably get a head start in the meantime 

Thanks!

**Comment by RossZhang-cpu** [comment]
Hi @yeikel, Could you kindly inform me of specific location where the incorrect hashset was implemented?

**Comment by yeikel** [comment]
> Hi @yeikel, Could you kindly inform me of specific location where the incorrect hashset was implemented? Thanks.

I think that you misunderstood this issue

The goal of the issue is not to fix something within this codebase but to create a rewrite recipe that will identify that pattern and apply the changes on any codebase

**Comment by RossZhang-cpu** [comment]
Hi @yeikel , Thanks for the clarification. Let me look into it.

**Comment by yeikel** [comment]
> Hi @yeikel , Thanks for the clarification. Let me look into it.

No problem. You can start here https://docs.openrewrite.org/concepts-and-explanations/recipes
